### PR TITLE
Update cuTENSOR installer for CUDA 12.x

### PR DESCRIPTION
**Pull request by kmaehashi**
Follow up #7284.

Found in https://github.com/cupy/cupy-release-tools/pull/322 (https://ci.preferred.jp/cupy-release-tools/120571/)

### CODE REVIEW

**Comment by leofang** [comment]
I don't understand, we use the same tarball at https://developer.download.nvidia.com/compute/cutensor/redist/libcutensor/linux-x86_64/libcutensor-linux-x86_64-1.6.2.3-archive.tar.xz both internally and externally (on conda-forge) for CUDA 10.2+. Nothing has changed compared to 1.6.1. What am I missing?

**Comment by kmaehashi** [comment]
Ah right, you are correct. It seems my tar file download was interrupted for some reason and I was only seeing 12 and 11 in the extracted directory...

**Review by leofang — APPROVED** [review]

File: cupyx/tools/install_library.py [hunk 1]
```diff
@@ -247,6 +247,8 @@ def install_lib(cuda, prefix, library):
         elif library == 'cutensor':
             if cuda.startswith('11.') and cuda != '11.0':
                 cuda = '11'
```
[the remark this answers is not in the document]
Comment (kmaehashi) [reply cupyx/tools/install_library.py:249]: I prefer the safer side here, as the env the installer runs and CuPy runs may be different (e.g. cluster with NFS-shared home directory).

**Review by asi1024 — APPROVED** [review]

**Pull request merged**